### PR TITLE
Initial Github Action

### PR DESCRIPTION
**Pull request by RanbirAulakh**
**Issue #, if available:** n/a

**Description of changes:**

- Removed existing buildspec.yml
- See below the diagram of how it works from start to finish
- Right now, Destroy stack is yet to be implemented, I will need to create a bash script to handle/check. For now, it does the deploy, build, and test. 

GithubActionsOSML

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+jobs:
+  CheckPendingWorkflow:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: ahmadnassri/action-workflow-queue@v1
+      with:
+        delay: 60000
+  BuildOSML:
+    runs-on: ubuntu-latest
```
Comment (RanbirAulakh) [inline .github/workflows/main.yml:25]: Forgot to add: `needs: CheckPendingWorkflow` before we can continue `BuildOSML`

Comment (RanbirAulakh) [reply .github/workflows/main.yml:26]: Added in next revision.

**Comment by RanbirAulakh** [comment]
I have bumped up the CodeBuild concurrent limit from 1 to 5 for `CodeBuild-TestGithubOSML`. I wonder if that will create a bottleneck if running 5 tests at the same time. Lets see if that succeeded and determine how long that took.

If running 1 test at a time for 5 jobs, it took about 14m 14s

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            TEST_IMAGE: "tile_tif"
+            TEST_MODEL: "aircraft"
+        with:
+          env-vars-for-codebuild: |
+            TEST_IMAGE,
+            TEST_MODEL
+          project-name: CodeBuild-TestGithubOSML
+          buildspec-override: .github/codebuild/test.yml
+  DestroyOSML:
+    needs: [RunSmallTifCenterpointTest, RunMetaNtfCenterpointTest, RunLargeTifFloodTest, RunTileNtfAircraftTest, RunTileTifAircraftTest]
```
Comment (RanbirAulakh) [inline .github/workflows/main.yml:145]: Going add `if: ${{ always() }}` if any of the deployment/test fails. We need to ensure that we can safely delete all the stacks and resources.

Comment (RanbirAulakh) [reply .github/workflows/main.yml:156]: Need to add `BuildOSML` to `needs:`

**Comment by RanbirAulakh** [comment]
Allows you to build and test OSML in either main or dev account depending the situation.

It can dynamically detect which aws_account to use. So basically
- <any branches> PR’d to dev branch, it will build and test in dev account
- dev branch is PR’d to main branch, it will build and test in main account

**Review by drduhe — COMMENTED** [review]
LGTM

**Review by drduhe — APPROVED** [review]

**Pull request merged**